deploy/template.yaml

```diff
@@ -1325,6 +1325,99 @@ Resources:
               Bool:
                 "aws:SecureTransport": true
 
+  #
+  # DynamoDB
+  #
+
+  SessionsDynamoDB:
+    Type: AWS::DynamoDB::Table
+    Properties:
+      AttributeDefinitions:
+        - AttributeName: "id"
+          AttributeType: "S"
+        - AttributeName: "subjectId"
+          AttributeType: "S"
+      BillingMode: PAY_PER_REQUEST
+      ContributorInsightsSpecification:
+        Enabled: true
+      GlobalSecondaryIndexes:
+        - IndexName: "subjectIdIndex"
+          KeySchema:
+            - AttributeName: "subjectId"
+              KeyType: "HASH"
+          Projection:
+            ProjectionType: KEYS_ONLY
+      KeySchema:
+        - AttributeName: "id"
+          KeyType: "HASH"
+      SSESpecification:
+        KMSMasterKeyId: !Ref DynamoDBKmsKey
+        SSEEnabled: true
+        SSEType: KMS
+      TableClass: STANDARD
+      TableName: !Sub ${AWS::StackName}-SessionsDynamoDB
+      Tags:
+        - Key: Product
+          Value: !Ref ProductTagValue
+        - Key: System
+          Value: Accounts
+        - Key: Environment
+          Value: !Ref Environment
+        - Key: Owner
+          Value: !Ref OwnerTagValue
+        - Key: Source
+          Value: !Ref SourceTagValue
+        - Key: Application
+          Value: Shared
+
+  DynamoDBKmsKey:
+    Type: AWS::KMS::Key
+    Properties:
+      Description: !Sub "KMS key used to protect DynamoDB data in ${AWS::StackName}"
+      Enabled: true
+      EnableKeyRotation: true
+      KeyPolicy:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS: !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            Action:
+              - kms:*
+            Resource: "*"
```

Comment:
🤔 could we have a bit of a chat about this one.
This seems kinda broad, I'm still a bit of a newbie to AWS permissions but this seems to be a broad KMS permission, applied to all resources. I suspect we could narrow that a bit.

Also not sure the ffect of L1383/L1384, but `root` on the account ID as a principle makes me 🤔

Reply:
Hi @huwd, as per documentation [here](https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html) by default AWS KMS key policies do not provide account permissions and we have therefore been providing root user access to all of our KMS resources to allow us to manage our KMS keys. That being said you may be right that having root access is too broad and instead we should potentially provide access to our admin roles to manage the keys. This affects all of our KMS related policies, not just this resource alone, so I'll create an enhancement ticket to further investigate what the best approach is and reflect on all KMS related key policies.

Reply:
So if you're happy to resolve this comment in scope of this work, I'll look into it from a broader KMS policy perspective.

Reply:
Just had a look at other repos and seems other teams are providing the same `root` permissions. Will investigate further.

Reply:
Hi @huwd did some investigating on the above and we'll need to keep the KMS key policy permissions as is, please see comments in ticket https://govukverify.atlassian.net/browse/ATB-155.